.github/workflows/build_and_run_test.yml

```diff
@@ -102,9 +102,52 @@ jobs:
         run: |
           make clean-docker-${{ matrix.chain }}
 
+  run-cargo-unit-tests:
+    runs-on: ubuntu-latest
+    needs: check-cargo-fmt
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          target: wasm32-unknown-unknown
+          default: true
+
+      - name: Run unittests
+        run: cargo test --release -p pallet-* --lib
```

Comment:
Will this run all unit tests of pallets with the name prefix "pallet-"?

Reply:
Also I ran this command locally and got an error `zsh: no matches found: pallet-*`, did you run it successfully?

Reply:
> Also I ran this command locally and got an error `zsh: no matches found: pallet-*`, did you run it successfully?

I believe it comes from the expansion bevaiour of `*` under zsh, maybe try this?
```
setopt nonomatch
```

Anyways on both Linux-bash and MacOS-bash I can get the correct behaviour, where unittests of all pallets will be run.

Reply:
Yes, it works after switching to bash.

However, it seems that here only the unit tests of modules starting with pallet- prefix are executed. I see your points to separate these with the runtime ones, but by doing this it makes CI execution even longer. Also, there are some packages like primitives which are excluded from running test, and these could be an issue later when we add any packages without the pallet prefix.

Reply:
> Yes, it works after switching to bash.
> 
> However, it seems that here only the unit tests of modules starting with pallet- prefix are executed. I see your points to separate these with the runtime ones, but by doing this it makes CI execution even longer. Also, there are some packages like primitives which are excluded from running test, and these could be an issue later when we add any packages without the pallet prefix.

Unittests are meant for pallets only, we could provide a list of pallet names for unittests if we have pallets without standard prefix.

Unittests/integration tests/ts-tests are running in parallel. Actually, ts-tests take the most of the time, so CI is not getting longer (or, it gets longer just because we have to do both litmus and litentry ts-tests and these can only be done in sequential because of dependencies in docker, it takes 3 min anyway).

Reply:
It seems it takes 20 min each according to the action run, but I am not sure if it can be boosted if we do unit test and integration test together. 

The current CI time is 52 min already, comparing to the previous time (20min+) it is doubled the time. We may consider to reduce the execution time by offload some non-dependent jobs (clippy, fmt) to github provided hosts, in order to boost the main jobs on our self-hosted machine. What do you think?

Reply:
It's not 

> It seems it takes 20 min each according to the action run, but I am not sure if it can be boosted if we do unit test and integration test together.
> 
> The current CI time is 52 min already, comparing to the previous time (20min+) it is doubled the time. We may consider to reduce the execution time by offload some non-dependent jobs (clippy, fmt) to github provided hosts, in order to boost the main jobs on our self-hosted machine. What do you think?

52min is really the exception where the sefl-hosted machine was round-robined to run CI on both `dev` branch and feature branch. If you check the longest path, it's 26min.
The normal CI process will take 25min, e.g. here: https://github.com/litentry/litentry-parachain/actions/runs/1733110429

Reply:
we **are** doing unittests and integrations in parallel, and not on our hosts. So we don't care about the resources for it as long as it's not the slowest part in the whole CI.

We could simply run `cargo test --release`, but the exact failure won't be available on the CI panel (e.g. you don't know if it unittest fails or integration-test fails.

Reply:
btw `clippy` is also run on the github host too